app/services/CategorisationService.scala

```diff
@@ -39,122 +39,87 @@ class CategorisationService @Inject() (
+                                   case Some(categorisationInfo) => Future.successful(categorisationInfo)
+                                   case _                        => Future.failed(new RuntimeException("Could not build categorisation info"))
+                                 }
+      updatedAnswers          <-
         Future.fromTry(
           request.userAnswers.set(
             RecordCategorisationsQuery,
             recordCategorisations.copy(records = recordCategorisations.records + (recordId -> categorisationInfo))
           )
         )
-      _                      <- sessionRepository.set(updatedAnswers)
+      _                       <- sessionRepository.set(updatedAnswers)
     } yield updatedAnswers
-  }
 
-  //TODO this will be refactored out in TGP-1600 but solves the immediate problem
-  def updateCategorisationWithUpdatedCommodityCode(
+  def updateCategorisationWithLongerCommodityCode(
     request: DataRequest[_],
     recordId: String
   )(implicit
     hc: HeaderCarrier
-  ): Future[UserAnswers] = {
-
-    val recordCategorisations =
-      request.userAnswers.get(RecordCategorisationsQuery).getOrElse(RecordCategorisations(Map.empty))
-
-    for {
-      newCommodityCode                 <- Future.fromTry(Try(request.userAnswers.get(CommodityUpdateQuery(recordId)).get))
-      getGoodsRecordResponse           <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
-      goodsNomenclature                <- ottConnector.getCategorisationInfo(
-                                            newCommodityCode.commodityCode,
-                                            request.eori,
-                                            request.affinityGroup,
-                                            Some(recordId),
-                                            getGoodsRecordResponse.countryOfOrigin,
-                                            LocalDate.now() //TODO where does DateOfTrade come from??
-                                          )
-      categorisationInfo               <- Future.fromTry(Try(CategorisationInfo.build(goodsNomenclature).get))
-      updatedAnswers                   <-
-        Future.fromTry(
-          request.userAnswers.set(
-            RecordCategorisationsQuery,
-            recordCategorisations.copy(records = recordCategorisations.records + (recordId -> categorisationInfo))
-          )
-        )
-      updatedAnswersCleanUpAssessments <-
-        Future.fromTry(cleanupOldAssessmentAnswers(updatedAnswers, recordId))
-      _                                <- sessionRepository.set(updatedAnswersCleanUpAssessments)
-    } yield updatedAnswersCleanUpAssessments
-  }
+  ): Future[UserAnswers] =
+    (for {
+      getGoodsRecordResponse <- goodsRecordsConnector.getRecord(eori = request.eori, recordId = recordId)
+      newCommodityCode       <- Future.fromTry(Try(request.userAnswers.get(LongerCommodityQuery(recordId)).get))
```

Comment:
Can we handle the exception here/not do .get

Reply:
The Future.fromTry handles it doesn't it? - it will return a Future.failed if it isn't defined

Reply:
I agree, I believe .get should be allowed in cases where it's handled safely